lib/pure/results.nim

```diff
@@ -0,0 +1,124 @@
+import typetraits, strutils
+
+type
+  Result*[T] = object
+    ## Either a value or an error.
+    case isSuccess*: bool:
+    of true:
+      value*: T
+    of false:
+      error*: ref Exception
+      backtrace*: BacktraceFrame
```

Comment:
Well ok, but the ``next`` pointer is not set in your code and I don't see how it would work without compiler support.